src/utils/deleteGuildRole.ts

```diff
@@ -0,0 +1,29 @@
+import { INTERNAL_SERVER_ERROR, ROLE_REMOVED } from "../constants/responses";
+import { DISCORD_BASE_URL } from "../constants/urls";
+import { env } from "../typeDefinitions/default.types";
+import createDiscordHeaders from "./createDiscordHeaders";
+
+export async function deleteGuildRole(
+  env: env,
+  roleId: string,
+  reason?: string
+) {
+  const deleteGuildRoleUrl = `${DISCORD_BASE_URL}/guilds/${env.DISCORD_GUILD_ID}/roles/${roleId}`;
+  const headers: HeadersInit = createDiscordHeaders({
+    token: env.DISCORD_TOKEN,
+    reason: reason,
```

Comment:
What happens with the reason value, can you share some context of this.

Reply:
It is passed to discord api in header and discord uses it for audit log.